quinn-proto/src/connection.rs

```diff
@@ -568,8 +576,7 @@ where
                 .expect("update not acknowledged yet")
                 .1
         };
-        self.io
-            .timer_start(TimerKind::KeyDiscard, start + self.pto() * 3);
+        self.timers[Timer::KeyDiscard] = Some(start + self.pto() * 3);
```

Comment:
Should we make the `timers[foo] = Some(..)` and `timers[foo] = None` a bit more abstract, along the lines of `timers.arm(foo, ..)`/`timers.reset(foo)` (bikeshed names here)?

Reply:
This worked out really nicely, good call.